sdks/java/core/src/main/java/org/apache/beam/sdk/util/ReduceFnRunner.java

```diff
@@ -661,12 +669,28 @@ private void clearAllState(
       reduceFn.clearState(renamedContext);
       watermarkHold.clearHolds(renamedContext);
       nonEmptyPanes.clearPane(renamedContext.state());
+      // These calls work irrespective of whether the window is active or not, but
+      // are unnecessary if the window is not active.
       triggerRunner.clearState(
           directContext.window(), directContext.timers(), directContext.state());
+      paneInfoTracker.clear(directContext.state());
```

Comment:
I have lots of no-state-left-behind tests in https://github.com/apache/incubator-beam/pull/139. Can we wait until then?